apps/admin_app/lib/admin_app_web/controllers/product_controller.ex

```diff
@@ -88,21 +89,58 @@ defmodule AdminAppWeb.ProductController do
     )
   end
 
-  def add_images(conn, %{"product_images" => product_images, "product_id" => id}) do
-    product =
-      id
-      |> String.to_integer()
-      |> ProductModel.get()
-      |> Repo.preload(:images)
+  defp preload_product_images(id) do
+    id
+    |> String.to_integer()
+    |> ProductModel.get()
```

Comment:
No need to convert to string id integer as ProductModel support string.
`Snitch.Tools.Helper.Query` module has been update to use string ids.